core/domain/query_jobs_one_off.py

```diff
@@ -0,0 +1,143 @@
+# coding: utf-8
+#
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs to execute and get result of a query."""
+import ast
+import datetime
+
+from core import jobs
+from core.domain import exp_services
+from core.domain import query_domain
+from core.platform import models
+
+(user_models, job_models, exp_models) = (
+    models.Registry.import_models(
+        [models.NAMES.user, models.NAMES.job, models.NAMES.exploration]))
+
+
+class QueryOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for excuting query with given query parameters.
+    For each user we check if he/she satisfies query criteria. If user satisfies
+    query criteria then yeilds a tuple of (query_id, user_id). Reducer function
+    stores all such satisfying user_ids in query result model."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_settings_model):
+        query_id = jobs.BaseMapReduceJobManager.get_mapper_param('query_id')
+        query = job_models.QueryModel.get(query_id)
+        query_parameters = query_domain.get_parameters_from_query(query)
+        user_id = user_settings_model.id
+        user_qualified = True
+
+        if user_id == query.submitter_id:
+            user_qualified = False
+
+        if user_settings_model.username == 'tmpsuperadm1n':
```

Comment:
Thanks for bringing this up. While writing back end tests, in addition to 4 users and submitter I created, one more user namely super admin gets created for each test. MR job considers that user as a normal user and includes it in final result. Which caused failure in back end tests.

Aside from that, later on (in production website), we will exclude some users from final result. Just like submitter of query. So I was meaning to ask you how we will remove such users ? (While writing back end tests, I did not see any function that identifies whether user is admin or not. So I just patched it up and I was gonnna ask about it to you here.)